src/main/java/net/sf/jabref/model/groups/SearchGroup.java

```diff
@@ -21,18 +21,26 @@
 
     public static final String ID = "SearchGroup:";
 
-    private final SearchQuery query;
+    private final GroupSearchQuery query;
 
     private static final Log LOGGER = LogFactory.getLog(SearchGroup.class);
 
+    private final String searchExpression;
+    private final boolean caseSensitive;
+    private final boolean regExp;
+
+
     /**
      * Creates a SearchGroup with the specified properties.
      */
     public SearchGroup(String name, String searchExpression, boolean caseSensitive, boolean regExp,
             GroupHierarchyType context) {
         super(name, context);
 
-        this.query = new SearchQuery(searchExpression, caseSensitive, regExp);
+        this.searchExpression = searchExpression;
```

Comment:
Checking again, it was to not having to provide get-methods to the new, minimal, `GroupSearchQuery` as that is only used here and the information might just as well stay on this level. (The reason for introducing `GroupSearchQuery` was to not move even more search stuff to model.)